scc/bls/bls_test.go

```diff
@@ -162,6 +162,18 @@ func TestPublicKey_String_IsHexEncoded(t *testing.T) {
 	require.Regexp(t, regexp, key.String())
 }
 
+func TestPublicKey_CanBeMarshalAndUnmarshal(t *testing.T) {
+	require := require.New(t)
+	key := PublicKey{}
+	data, err := key.MarshalJSON()
+	require.NoError(err)
+
+	var key2 PublicKey
+	err = key2.UnmarshalJSON(data)
+	require.NoError(err)
+	require.Equal(key, key2)
+}
+
```

Comment:
There should be a test like this added (also for the signature) to verify that the encoding is indeed using the desired format:
```
func TestPublicKey_JsonEncodingUsesHexFormat(t *testing.T) {
	regexp := regexp.MustCompile(`^"0x[0-9a-f]{96}"$`)
	key := NewPrivateKey().PublicKey()
	data, err := json.Marshal(key)
	require.NoError(t, err)
	require.Regexp(t, regexp, string(data))
}
```